// Copyright 2018 The Hypebot Authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package hypebot;

message User {
  // Unique identifier for the user. Not necessarily human readable. Used by the
  // interface to send messages to a specific user and by the storage system to
  // keep track of your tendies.
  string user_id = 1;

  // Human readable display name for the user. Not necessarily unique, but
  // should be used when rendering the user's name.
  string display_name = 2;

  // True if the user is a bot, False if it is a human. Defaults to human on
  // purpose since almost all Users are human.
  bool bot = 3;
}
